src/modules/common/mentors.service.ts

```diff
@@ -78,20 +82,49 @@ export class MentorsService {
       .sort({ createdAt: 'desc' })
       .exec();
 
+    const mentorsForCurrentUser = new Set();
+
+    // determine if we have a logged in user
+    if (isLoggedIn) {
+      const user = await this.userModel
+        .findOne({ auth0Id: userAuth0Id })
+        .exec();
+
+      if (user) {
+        // find all of their mentorships
+        const mentorships: Mentorship[] = await this.mentorshipModel
+          .find({ mentee: user._id })
+          .exec();
+
+        // flatten the mentors into a set
+        mentorships.forEach(mentorship =>
+          mentorsForCurrentUser.add(mentorship.mentor.toString()),
+        );
+      }
+    }
+
     return {
-      mentors: mentors.map(mentor => ({
-        _id: mentor._id,
-        available: mentor.available,
-        spokenLanguages: mentor.spokenLanguages,
-        tags: mentor.tags,
-        name: mentor.name,
-        avatar: mentor.avatar,
-        title: mentor.title,
-        description: mentor.description,
-        country: mentor.country,
-        createdAt: mentor.createdAt,
-        channels: mentor.available ? mentor.channels : [],
-      })),
+      mentors: mentors.map(mentor => {
+        // channels are only visible to a mentee if they have a mentorship with the mentor
+        let showChannels = false;
+        if (mentorsForCurrentUser.has(mentor._id.toString())) {
+          showChannels = true;
+        }
```

Comment:
This can be simplified to `const showChannels = mentorsForCurrentUser.has(mentor._id.toString())`.

Reply:
Great call.  In an early iteration there was more logic in here.  Now that it's gone, this code is unnecessarily verbose.  Thanks for pointing it out!